create_statement_creator/athena_schema.py

```diff
@@ -36,16 +37,39 @@ def create_athena_column_schema(avro_schema) -> str:
         return f'struct<{field_schema_concatenated}>'
 
     elif type(avro_schema) == UnionSchema:
-        # pick the first schema which is not null
         union_schemas_not_null = [s for s in avro_schema.schemas if s.type != 'null']
-        if len(union_schemas_not_null) > 0:
+        contains_null_schema = len([s for s in avro_schema.schemas if s.type == 'null']) > 0
+        if contains_null_schema and len(union_schemas_not_null) == 1:
             return create_athena_column_schema(union_schemas_not_null[0])
-        else:
+        elif contains_null_schema and len(union_schemas_not_null) == 0:
             raise Exception('union schemas contains only null schema')
 
+        union_schemas = []
+        i = 0
+        for schema in union_schemas_not_null:
+            union_schemas.append(f'member{i}:{create_athena_column_schema(schema)}')
+            i += 1
+        members_concatenated = ','.join(union_schemas)
+        return f'struct<{members_concatenated}>'
+
     elif type(avro_schema) in [EnumSchema, FixedSchema]:
         return 'string'
 
+    elif type(avro_schema) in [BytesDecimalSchema, FixedDecimalSchema]:
+        return f'decimal({avro_schema.precision},{avro_schema.scale})'
+
+    elif type(avro_schema) == DateSchema:
+        return 'date'
+
+    elif type(avro_schema) == TimeMillisSchema:
+        return 'int'
+
+    elif type(avro_schema) == TimeMicrosSchema:
+        return 'long'
+
+    elif type(avro_schema) in [TimestampMillisSchema, TimestampMicrosSchema]:
+        return 'timestamp'
+
```

Comment:
Maybe one can extend the example schema in https://github.com/dataqube-GmbH/avro2athena/tree/master/example_schemas by example of these data types?

Reply:
Sure, I'll have a look at that at some point.